tests/std/tests/P0429R9_flat_map/test.cpp

```diff
@@ -265,7 +268,40 @@ void test_pointer_to_incomplete_type() {
     flat_map<MyType<Incomplete>, shared_ptr<MyType<Incomplete>>> fmap;
 }
 
+// Test MSVC STL-specific SCARY-ness
+namespace scary_test {

```

Comment:
Suggestion: This test is visually extremely noisy, and I think it could be much more readable with some typedefs extracted.